src/components/forms/PractitionerForm/UserIdSelect/index.tsx

```diff
@@ -81,27 +87,28 @@ export const UserIdSelect = (props: Props) => {
   };
 
   /** Pulls all users data */
-  const loadUsers = async (service: typeof OpenSRPService = OpenSRPService) => {
+  const loadUsers = async () => {
     let filterParams = {
       page_size: USERS_REQUEST_PAGE_SIZE,
+      source: OPENSRP_KEYCLOAK_PARAM,
       start_index: 0,
     };
-    const serve = new service(OPENSRP_USERS_ENDPOINT);
+    const serve = new props.serviceClass(OPENSRP_USERS_ENDPOINT);
     const allUsers = [];
-    let response: UserResponse;
+    let response: User[];
     do {
       response = await serve.list(filterParams).catch(err => {
         displayError(err);
       });
-      allUsers.push(...response.results);
+      allUsers.push(...response);
 
       // modify filter params to point to next page
-      const responseSize = response.results.length;
+      const responseSize = response.length;
       filterParams = {
         ...filterParams,
         start_index: filterParams.start_index + responseSize,
       };
-    } while (thereIsNextPage(response));
+    } while (filterParams.start_index < response.length);
```

Comment:
I am trying to think about this:
- seems like this will never be true:
```
- assuming there is more pages of data, then responseSize < len(totalData)
- but seems like we are using responseSize to know if there is more data.
- set start_index = 0
- say responseSize = 50
- set start_index = 50
- now start_index < responseSize => 50 < 50 => which is false
```

Reply:
updated this logic via https://github.com/onaio/reveal-frontend/pull/1166/commits/a01e152355358f8c40a89d8d7d52a059c5e1a34a